In [97]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
import seaborn as sns


# TASK 1

In [98]:

# reading the file for data and creating a dataframe
Grc_df = pd.read_csv("./dataset/Grocery_dataset.csv")


# gives number of (rows, columns)
print(f"Number of rows and columns respectively: {Grc_df.shape}")

# Printing first 10 rows
print(Grc_df.head(10))

# printing last 10 rows
print(Grc_df.tail(10))


Number of rows and columns respectively: (5000, 12)
  Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15        9.300          Low Fat         0.016047   
1           DRC01        5.920          Regular         0.019278   
2           FDN15       17.500          Low Fat         0.016760   
3           FDX07       19.200          Regular         0.000000   
4           NCD19        8.930          Low Fat         0.000000   
5           FDP36       10.395          Regular         0.000000   
6           FDO10       13.650          Regular         0.012741   
7           FDP10          NaN          Low Fat         0.127470   
8           FDH17       16.200          Regular         0.016687   
9           FDU28       19.200          Regular         0.094450   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6

# TASK 2 

In [99]:
Grc_df.isnull().sum()

Item_Identifier                 0
Item_Weight                   818
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  1439
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

Item_weight and Outlet_size are the only columns with missing values. Item_weight has 818 missing values while Outlet_size has 1439.

Im going to use 'dropna() method' that was used during the lab to remove the rows with missing values and then i will reset the index so the dataset stays clean.

In [100]:
Grc_df.dropna(inplace=True)
Grc_df.reset_index(drop=True, inplace=True)   
print('Shape after dropna:', Grc_df.shape)   # (2743, 12)
Grc_df.isnull().sum()  

Shape after dropna: (2743, 12)


Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

# TASK 5

In [101]:
# dropping columns with index 0 and 6, then naming new dataframe Grc_new_df
Grc_new_df = Grc_df.drop(columns=Grc_df.columns[[0, 6]])

# TASK 6

In [102]:
Grc_new_df['Outlet_Type'].unique()
# checking  if the values actually exist in that column, so i know exactly which strings to filter on

<StringArray>
['Supermarket Type1', 'Supermarket Type2']
Length: 2, dtype: str

In [103]:
# splitting Grc_new_df into two supermarket types DataFrames
SupType_1 = Grc_new_df[Grc_new_df['Outlet_Type'] == 'Supermarket Type1'].copy()
SupType_2 = Grc_new_df[Grc_new_df['Outlet_Type'] == 'Supermarket Type2'].copy()

# printing the shapes to confirm the split
print('SupType_1 shape:', SupType_1.shape)
print('SupType_2 shape:', SupType_2.shape)


SupType_1 shape: (2217, 10)
SupType_2 shape: (526, 10)


# TASK 9

In [104]:
Grc_Concat_df = Grc_Concat_df.set_index(
  ['Outlet_Size', 'Outlet_Location_Type']
)

NameError: name 'Grc_Concat_df' is not defined